File: integrations/aws-strands/error-code-corpus/typescript/code_through_classifier.ts
/** A code reached through a top-level classifier the site calls. */
import { EventType, type BaseEvent } from "@ag-ui/core";

function classify(error: unknown): string {
  if (error instanceof TypeError) return "CLASSIFIER_ADAPTER_BUG";
  return "CLASSIFIER_STRANDS_ERROR";
}

export function emit(error: unknown): BaseEvent {
  return {
    type: EventType.RUN_ERROR,
    message: "Resolved through a classifier.",
    code: classify(error),
  };
}

File: integrations/aws-strands/error-code-corpus/python/code_through_classifier.py
"""A code reached through a module-level classifier the site calls."""

from ag_ui.core import EventType, RunErrorEvent


def classify(error: Exception) -> str:
    if isinstance(error, KeyError):
        return "CLASSIFIER_ADAPTER_BUG"
    return "CLASSIFIER_STRANDS_ERROR"


def emit(error: Exception) -> RunErrorEvent:
    return RunErrorEvent(
        type=EventType.RUN_ERROR,
        message="Resolved through a classifier.",
        code=classify(error),
    )
